src/app/components/home-page/__tests__/HomePage.spec.tsx

```diff
@@ -10,11 +10,28 @@ jest.mock('../../../../shared/hooks/useMatchMedia');
 
 (useReducedMotion as jest.Mock).mockReturnValue(true);
 
+jest.mock('../NonCritical', () => ({
+  __esModule: true,
+  default: () => '{{ NonCritical }}',
+}));
+
+jest.mock('../../../../shared/components/layouts/UniProtFooter', () => ({
+  __esModule: true,
+  default: () => '{{ UniProtFooter }}',
+}));
+
 let rendered: ReturnType<typeof customRender>;
 
 describe('HomePage component', () => {
-  beforeEach(() => {
+  beforeEach(async () => {
     rendered = customRender(<HomePage />);
+
+    await waitFor(() => {
```

Comment:
This is part of the setup code for the tests below, we're not testing anything here, just waiting for the page to render fully before running the tests below otherwise we get loadings components and the `act()` warning because the page updates after the tests have finish running.
That being said, these are still `getBy` calls that would make the tests fail if there's a change and these 3 things don't render before the default timeout

Reply:
For me it's not clear when the `waitFor` knows the `getBy`'s (or whatever the body of the of the callback) have found what it wants. I'm not saying what you have is wrong, just that I don't understand how it is working. In my head the current code looks like `f` below while I was expecting `g`

```
const f = () => {
    1+1
}
f() // undefined

const g = () => {
    return 1+1
}
g() // 2

```

Reply:
Possibly when something is not found an exception is thrown and that's how the `waitFor` knows not to proceed.